src/sync/mutex.go

```diff
@@ -37,6 +37,19 @@ func (m *Mutex) Unlock() {
 	}
 }
 
+// TryLock tries to lock m and reports whether it succeeded.
+//
+// Note that while correct uses of TryLock do exist, they are rare,
+// and use of TryLock is often a sign of a deeper problem
+// in a particular use of mutexes.
+func (m *Mutex) TryLock() bool {
+	if m.locked {
```

Comment:
It wouldn't surprise me if we needed calls to volatile loads here: https://github.com/tinygo-org/tinygo/blob/release/src/runtime/volatile/volatile.go

Reply:
:thinking: Wouldn't a memory barrier be better suited for this use case? Even then, tinygo is single threaded so we should have consistent ordering?

Reply:
I'm more concerned about LLVM removing loads/unloads it doesn't think are important.  Yes, TInyGo is single threaded so memory barriers are less interesting.

Reply:
I've added volatile loads/stores as requested!